### PR TITLE
Handle POST 200 level requests

### PR DESCRIPTION
**Pull request by nadiacollado**
- Handle POST 200 level requests
- Refactors `setBody` in `RequestParser` to use `read()` instead of `readLine()`

### CODE REVIEW

File: src/main/java/httpserver/request/RequestParser.java [hunk 1]
```diff
@@ -19,6 +16,7 @@ public Request parse(InputStream httpRequest) throws IOException {
 
        setStartLine(startLine, requestBuilder);
        setHeaders(requestBuilder);
+       setBody(contentLength, requestBuilder);
        return requestBuilder.build();
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/request/RequestParser.java:20]: This reads a lot better and doesnt hide the body creation in the header section. Nice refactor.

File: src/main/java/httpserver/request/RequestParser.java [hunk 2]
```diff
@@ -39,21 +37,22 @@ public void setHeaders(RequestBuilder requestBuilder) throws IOException {
...
 
     public void setBody(String contentLength, RequestBuilder requestBuilder) throws IOException {
-        if (contentLength.equals("0")) return;
-        String bodyLine;
-        StringBuilder body = new StringBuilder();
-        while ((bodyLine = reader.readLine()) != null) {
-            body.append(bodyLine + "\n");
+        if (contentLength != null && contentLength != "0") {
+            String parsedBod = null;
+            int bodyLength = Integer.parseInt(contentLength);
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/request/RequestParser.java:49]: Is this a typo for parsedBody?

**Review by ranizilpelwar — APPROVED** [review]
Nice Job! So happy you got the request body reading working!

File: src/main/java/httpserver/request/RequestParser.java (hunk 2 above)
Comment (nadiacollado) [reply src/main/java/httpserver/request/RequestParser.java:49]: Yes, will fix!

**Pull request merged**